ros/src/tl_detector/tl_detector.py

```diff
@@ -21,19 +22,26 @@ class TLDetector(object):
     def __init__(self):
         rospy.init_node('tl_detector')
 
-        model_path = os.path.abspath(rospy.get_param('~inference_model'))
+        tl_classifier_class = rospy.get_param('~inference_class')
 
         self.pose = None
         self.waypoints = None
         self.camera_image = None
         self.lights = []
         self.tl_waypoints_idx = []
 
+        self.harvest_images = False
+        if self.harvest_images:
+            if not (os.path.exists("./tl_images")):
+                os.mkdir("./tl_images")
+            self.debug_image_count = 0
+
         config_string = rospy.get_param("/traffic_light_config")
         self.config = yaml.load(config_string)
 
         self.bridge = CvBridge()
-        self.light_classifier = TLClassifier(model_path)
+        klass = globals()[tl_classifier_class]
```

Comment:
Is it possible to do it in one line?

        self.light_classifier = globals()[tl_classifier_class]()

Maybe you can add a comment explaining that's how you instantiate an object from a variable-named class, or put a link to a reference about it. At least this wasn't trivial to me, but if you think it's pretty standard Python then I'm fine without the link :)